clouddrift/analysis.py

```diff
@@ -1198,3 +1208,239 @@ def unpack_ragged(
+                "relative_vorticity must be a float or the same shape as time, lon, and lat."
+            )
+
+    # convert time interval of data in seconds
+    # dt = np.diff(time)[0] / np.timedelta64(1000000000, "ns")  # seconds no unit
+
+    # wavelet parameters
+    gamma = 3
+    density = 16  # JML uses 16
+    beta = wavelet_duration**2 / gamma
+
+    if isinstance(latitude, xr.DataArray):
+        latitude = latitude.to_numpy()
+    if isinstance(longitude, xr.DataArray):
+        longitude = longitude.to_numpy()
+
+    # Instantaneous absolute frequency of oscillations along trajectory in radian per second
+    cor_freq = np.abs(
+        coriolis_frequency(latitude) + relative_vorticity * np.sign(latitude)
+    )
+    cor_freq_max = np.max(cor_freq * 1.05)
+    cor_freq_min = np.max(
+        [np.min(cor_freq * 0.95), 2 * np.pi / (time_step * data_length)]
+    )
+
+    # logarithmically distributed frequencies for wavelet analysis
+    radian_frequency = morse_logspace_freq(
+        gamma,
+        beta,
+        data_length,
+        (0.05, cor_freq_max * time_step),
+        (5, cor_freq_min * time_step),
+        density,
+    )  # frequencies in radian per unit time
+
+    # wavelet transform on a sphere: (spheretrans in jLab)
+    # unwrap longitude recasted in [0,360)
+    longitude_unwrapped = np.unwrap(recast_lon360(longitude), period=360)
+
+    # convert lat/lon to Cartesian coordinates x, y , z
+    x, y, z = spherical_to_cartesian(longitude_unwrapped, latitude)
+
+    # wavelet transform of x, y, z
+    wavelet, _ = morse_wavelet(data_length, gamma, beta, radian_frequency)
+    wx = wavelet_transform(x, wavelet)
+    wy = wavelet_transform(y, wavelet)
+    wz = wavelet_transform(z, wavelet)
+
+    # x, y, z minus the transforms converted back to lat and lon
+    # dum = np.broadcast(x,wx)
+    # out1 = np.empty(dum.shape)
+    # out1.flat = [u-np.real(v) for (u,v) in dum]
+    # dum = np.broadcast(y,wy)
+    # out2 = np.empty(dum.shape)
+    # out2.flat = [u-np.real(v) for (u,v) in dum]
+    # dum = np.broadcast(z,wz)
+    # out3 = np.empty(dum.shape)
+    # out3.flat = [u-np.real(v) for (u,v) in dum]
+    # longitude_new, latitude_new = cartesian_to_spherical(
+    #     out1, out2, out3
+    # )
+    longitude_new, latitude_new = cartesian_to_spherical(
+        x - np.real(wx), y - np.real(wy), z - np.real(wz)
+    )
+
+    # convert transforms to horizontal displacements on tangent plane
+    wxh, wyh = cartesian_to_tangentplane(wx, wy, wz, longitude_new, latitude_new)
+
+    # rotary wavelet transforms to select inertial component; need to divide by sqrt(2)
+    wp = (wxh + 1j * wyh) / np.sqrt(2)
+    wn = (wxh - 1j * wyh) / np.sqrt(2)
+
+    # find the values of radian_frequency/dt that most closely match cor_freq
+    # is there a better way to do this?
+    frequency_bins = [
+        np.argmin(np.abs(cor_freq[i] - radian_frequency / time_step))
+        for i in np.arange(0, data_length)
```

Comment:
I think this can be:

```suggestion
        for i in range(data_length)
```